presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestGroupInnerJoinsByConnector.java

```diff
@@ -0,0 +1,531 @@
+    private static final PlanBuilder PLAN_BUILDER = new PlanBuilder(TEST_SESSION, new PlanNodeIdAllocator(), METADATA);
+
+    @Test
+    public void testDoesNotPushDownOuterJoin()
+    {
+        String catalogName = "test_catalog";
+        String connectorName = "test_catalog";
+        ConnectorId connectorId = new ConnectorId(catalogName);
+
+        MockConnectorFactory.Builder builder = MockConnectorFactory.builder();
+        ImmutableSet<ConnectorCapabilities> connectorCapabilities = immutableEnumSet(SUPPORTS_JOIN_PUSHDOWN);
+        MockConnectorFactory mockConnectorFactory = builder.withListSchemaNames(connectorSession -> ImmutableList.of("test_schema"))
+                .withConnectorCapabilities(connectorCapabilities)
+                .withListTables((connectorSession, schemaNameOrNull) ->
+                        ImmutableList.of(
+                                new SchemaTableName("test_schema", "test_view"),
+                                new SchemaTableName("test_schema", "another_table"))).build();
+        Connector testConnector = mockConnectorFactory.create(catalogName, ImmutableMap.of(), new TestingConnectorContext());
+
+        CatalogManager catalogManager = new CatalogManager();
+        catalogManager.registerCatalog(new Catalog(
+                catalogName,
+                connectorId,
+                testConnector,
+                createInformationSchemaConnectorId(connectorId),
+                testConnector,
+                createSystemTablesConnectorId(connectorId),
+                testConnector));
+
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Metadata metadata = createTestMetadataManager(transactionManager, new FeaturesConfig(), new FunctionsConfig());
+
+        TransactionId transactionId = transactionManager.beginTransaction(false);
+        Session session = testSessionBuilder().setTransactionId(transactionId).build();
+
+        ImmutableMap.Builder<ColumnHandle, Domain> domains = new ImmutableMap.Builder<>();
+        domains.put(new InformationSchemaColumnHandle("table_schema"), Domain.singleValue(VARCHAR, Slices.utf8Slice("test_schema")));
+        Constraint<ColumnHandle> constraint = new Constraint<>(TupleDomain.withColumnDomains(domains.build()));
+
+        InformationSchemaMetadata informationSchemaMetadata = new InformationSchemaMetadata(catalogName, metadata);
+        informationSchemaMetadata.getTableLayouts(
+                createNewSession(transactionId),
+                new InformationSchemaTableHandle(catalogName, "information_schema", "views"),
+                constraint,
+                Optional.empty());
+
+        VariableReferenceExpression left = newBigintVariable("a1");
+        VariableReferenceExpression right = newBigintVariable("a2");
+        EquiJoinClause joinClause = new EquiJoinClause(left, right);
+
+        PlanNode plan = output(join(FULL,
```

Comment:
Join pushdown will also not work for LEFT and RIGHT joins, can we add those assertions too ?